steps/generate_dataset_from_htme.py

```diff
@@ -1002,10 +1003,46 @@ def get_collection(collection_name):
     return collection_name.replace("db.", "", 1).replace(".", "/").replace("-", "_")
 
 
-def get_collections(secrets_response, args):
+def get_failed_collection_names(correlation_id, dynamodb_client) -> list:
+    failed_collections_response = dynamodb_client.query(
+        TableName=TABLE_NAME,
+        KeyConditionExpression=f"CorrelationId = :cid",
+        FilterExpression="begins_with(ADGStatus, :status)",
+        ExpressionAttributeValues={
+            ":cid": {"S": correlation_id},
+            ":status": {"S": "Failed"},
+        }
+    )
+    collections = [
+        failed_collection["CollectionName"]["S"]
+        for failed_collection in failed_collections_response["Items"]
+    ]
+    return collections
+
+
+def get_collections(secrets_response, args, dynamodb_client):
     try:
         collections = secrets_response["collections_all"]
         collections = {k: v for k, v in collections.items()}
+
+        if args.failed_collections_only is True:
+            failed_collections = get_failed_collection_names(
+                args.correlation_id,
+                dynamodb_client
+            )
+
+            failed_collections_to_process = {}
+
+            for k, v in collections.items():
+                if k in failed_collections:
+                    failed_collections_to_process[k] = v
+                    the_logger.warning(f"{k} is a failed collection, will be processed")
+                else:
+                    the_logger.warning(
+                        f"{k} is not a failed collection, will not be processed"
+                    )
+            collections = failed_collections_to_process
+
```

Comment:
It might be good to have a check somewhere that says, if length of collections is 0 then no failed collections to rerun. We presume that if the failed_collections_only flag is used, there are failed collections to rurun, which means to me a warning messages would be appropriate to signify to the user that no failed collections were found. Thoughts?

Reply:
Added